### PR TITLE
Move from FindBugs to SpotBugs 3.1.11 (#189)

### PR DESCRIPTION
**Pull request by SpOOnman**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 ## Usage
 
-Sputnik runs Checkstyle, PMD, FindBugs, CodeNarc, JSHint (or JSLint), TSLint, Ktlint, Detekt and Sonar only on files affected by Gerrit's patchset. It collects all violations and report them back to Gerrit or Stash.
+Sputnik runs Checkstyle, PMD, SpotBugs, CodeNarc, JSHint (or JSLint), TSLint, Ktlint, Detekt and Sonar only on files affected by Gerrit's patchset. It collects all violations and report them back to Gerrit or Stash.
```
Comment (alien11689) [inline README.md]: Add information `formerly known as FindBugs`

File: README.md [hunk 2]
```diff
@@ -27,9 +27,9 @@ checkstyle.configurationFile=sun_checks.xml
 checkstyle.propertiesFile=
 pmd.enabled=true
 pmd.ruleSets=rulesets/java/android.xml,rulesets/java/basic.xml
-findbugs.enabled=true
```
Comment (alien11689) [inline README.md]: Add comment or information that findbug properties still work

File: build.gradle [hunk 3]
```diff
@@ -95,9 +96,8 @@ dependencies {
         exclude group: 'xml-apis'
     }
 
-    // FindBugs dependencies
-    compile 'com.google.code.findbugs:findbugs:3.0.1'
-    compile 'xml-apis:xml-apis:1.0.b2' // this version is needed by FindBugs on runtime
```
Comment (alien11689) [inline build.gradle]: And now it isn't?

File: src/main/java/pl/touk/sputnik/processor/spotbugs/SpotBugsProcessor.java [hunk 4]
```diff
@@ -107,15 +107,21 @@ private IClassScreener createClassScreener(@NotNull Review review) {
 
     @Nullable
     private String getIncludeFilterFilename() {
-        String includeFilterFilename = config.getProperty(GeneralOption.FINDBUGS_INCLUDE_FILTER);
-        log.info("Using FindBugs include filter file {}", includeFilterFilename);
+        String includeFilterFilename = config.getProperty(GeneralOption.SPOTBUGS_INCLUDE_FILTER);
+        if (StringUtils.isBlank(includeFilterFilename)) {
+            includeFilterFilename = config.getProperty(GeneralOption.FINDBUGS_INCLUDE_FILTER);
```
Comment (alien11689) [inline src/main/java/pl/touk/sputnik/processor/spotbugs/SpotBugsProcessor.java]: I do not like reassigning variable values...

File: src/main/java/pl/touk/sputnik/processor/spotbugs/SpotBugsProcessor.java [hunk 5]
```diff
@@ -107,15 +107,21 @@ private IClassScreener createClassScreener(@NotNull Review review) {
...
         return includeFilterFilename;
     }
 
     @Nullable
     private String getExcludeFilterFilename() {
-        String excludeFilterFilename = config.getProperty(GeneralOption.FINDBUGS_EXCLUDE_FILTER);
-        log.info("Using FindBugs exclude filter file {}", excludeFilterFilename);
+        String excludeFilterFilename = config.getProperty(GeneralOption.SPOTBUGS_EXCLUDE_FILTER);
+        if (StringUtils.isBlank(excludeFilterFilename)) {
+            excludeFilterFilename = config.getProperty(GeneralOption.FINDBUGS_EXCLUDE_FILTER);
```
Comment (alien11689) [inline src/main/java/pl/touk/sputnik/processor/spotbugs/SpotBugsProcessor.java]: I do not like reassigning variable values...

File: src/test/java/pl/touk/sputnik/engine/ProcessorBuilderTest.java [hunk 6]
```diff
@@ -23,7 +23,7 @@ public void shouldNotBuildAnyProcessor() {
     public void shouldBuildDisabledProcessors() {
         Configuration config = new ConfigurationSetup().setUp(ImmutableMap.of(
                 GeneralOption.CHECKSTYLE_ENABLED.getKey(), "false",
-                GeneralOption.FINDBUGS_ENABLED.getKey(), "false",
+                GeneralOption.SPOTBUGS_ENABLED.getKey(), "false",
```
Comment (alien11689) [inline src/test/java/pl/touk/sputnik/engine/ProcessorBuilderTest.java]: IMHO both flags should work

File: README.md (hunk 2 above)
Comment (SpOOnman) [reply README.md]: Done

File: build.gradle (hunk 3 above)
Comment (SpOOnman) [reply build.gradle]: I've run integration tests and it runs fine. SpotBugs has upgraded to JDK 8 and I think that's why we don't need this backward dependency.

File: src/main/java/pl/touk/sputnik/processor/spotbugs/SpotBugsProcessor.java (hunk 4 above)
Comment (SpOOnman) [reply src/main/java/pl/touk/sputnik/processor/spotbugs/SpotBugsProcessor.java]: Me neither but it's reasonable here, WDYT?

File: src/main/java/pl/touk/sputnik/processor/spotbugs/SpotBugsProcessor.java (hunk 5 above)
Comment (SpOOnman) [reply src/main/java/pl/touk/sputnik/processor/spotbugs/SpotBugsProcessor.java]: Me neither but it's reasonable here, WDYT?

File: src/test/java/pl/touk/sputnik/engine/ProcessorBuilderTest.java (hunk 6 above)
Comment (SpOOnman) [reply src/test/java/pl/touk/sputnik/engine/ProcessorBuilderTest.java]: Good catch! I've added a few tests.

**Pull request merged**